src/masternodes/mn_rpc.cpp

```diff
@@ -1558,32 +2005,675 @@ UniValue accounttoutxos(const JSONRPCRequest& request) {
+                                                          "\\\"ownerAddress\\\":\\\"Address\\\""
+                                                            "}\" \"[{\\\"txid\\\":\\\"id\\\",\\\"vout\\\":0}]\" ")
                },
     }.Check(request);
 
-    /// @todo is it important to restrict with `Params().GetConsensus().AMKHeight` ??
+    if (pwallet->chain().isInitialBlockDownload()) {
+        throw JSONRPCError(RPC_CLIENT_IN_INITIAL_DOWNLOAD, "Cannot create transactions while still in Initial Block Download");
+    }
 
-    UniValue ret(UniValue::VOBJ);
+    RPCTypeCheck(request.params, {UniValue::VOBJ, UniValue::VARR}, true);
 
-    LOCK(cs_main);
-    for (auto kv : Params().GetConsensus().nonUtxoBlockSubsidies) {
-        ret.pushKV(GetCommunityAccountName(kv.first), ValueFromAmount(pcustomcsview->GetCommunityBalance(kv.first)));
+    std::string tokenA, tokenB, pairSymbol;
+    CAmount commission = 0; // !!!
+    CScript ownerAddress;
+    bool status = true; // default Active
+    UniValue metadataObj = request.params[0].get_obj();
+    if (!metadataObj["tokenA"].isNull()) {
+        tokenA = metadataObj["tokenA"].getValStr();
+    }
+    if (!metadataObj["tokenB"].isNull()) {
+        tokenB = metadataObj["tokenB"].getValStr();
+    }
+    if (!metadataObj["commission"].isNull()) {
+        commission = AmountFromValue(metadataObj["commission"]);
+    }
+    if (!metadataObj["status"].isNull()) {
+        status = metadataObj["status"].getBool();
+    }
+    if (!metadataObj["ownerAddress"].isNull()) {
+        ownerAddress = DecodeScript(metadataObj["ownerAddress"].getValStr());
+    }
+    if (!metadataObj["pairSymbol"].isNull()) {
+        pairSymbol = metadataObj["pairSymbol"].getValStr();
     }
 
-    return ret;
+    DCT_ID idtokenA, idtokenB;
+    {
+        LOCK(cs_main);
+
+        auto token = pcustomcsview->GetTokenGuessId(tokenA, idtokenA);
+        if (!token)
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "TokenA was not found");
+
+        auto token2 = pcustomcsview->GetTokenGuessId(tokenB, idtokenB);
+        if (!token2)
+            throw JSONRPCError(RPC_INVALID_ADDRESS_OR_KEY, "TokenB was not found");
+    }
+
+    CPoolPairMessage poolPairMsg;
+    poolPairMsg.idTokenA = idtokenA;
+    poolPairMsg.idTokenB = idtokenB;
+    poolPairMsg.commission = commission;
+    poolPairMsg.status = status;
+    poolPairMsg.ownerAddress = ownerAddress;
+
+    CDataStream metadata(DfTxMarker, SER_NETWORK, PROTOCOL_VERSION);
+    metadata << static_cast<unsigned char>(CustomTxType::CreatePoolPair)
+             << poolPairMsg << pairSymbol;
+
+    CScript scriptMeta;
+    scriptMeta << OP_RETURN << ToByteVector(metadata);
+
+    CMutableTransaction rawTx;
+    rawTx.vout.push_back(CTxOut(0, scriptMeta));
+
+    for(std::set<CScript>::iterator it = Params().GetConsensus().foundationMembers.begin(); it != Params().GetConsensus().foundationMembers.end() && rawTx.vin.size() == 0; it++)
```

Comment:
Why not use `for (auto& it : Params().GetConsensus().foundationMembers)`

Reply:
Cause "&& rawTx.vin.size() == 0" there in 'for' clause. 
and it can't deduce proper type here on simple "auto&" or "auto const &". "auto" works, but it create (i think) an object copy here

Reply:
I'll try to refactor it to improve readness later, but there are moorre places to refactor

Reply:
Can put the checking `rawTx.vin.size() == 0` inside the loop, then it is easier to read, because this line is too long, at first glance i didn't see `rawTx.vin.size() == 0` is there.

Reply:
Ok, but later. All "foundation auth" cases should be refactored at once, with distinct method